example/game/src/main/java/no/nordicsemi/android/ble/example/game/client/view/ClientScreen.kt

```diff
@@ -0,0 +1,136 @@
+package no.nordicsemi.android.ble.example.game.client.view
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import no.nordicsemi.android.ble.example.game.R
+import no.nordicsemi.android.ble.example.game.client.viewmodel.ClientViewModel
+import no.nordicsemi.android.ble.example.game.quiz.view.ShowResult
+import no.nordicsemi.android.ble.example.game.quiz.view.AskPlayersName
+import no.nordicsemi.android.ble.ktx.state.ConnectionState
+import no.nordicsemi.android.common.navigation.NavigationManager
+import no.nordicsemi.android.common.permission.RequireBluetooth
+import no.nordicsemi.android.common.theme.view.NordicAppBar
+
+@Composable
+fun ClientScreen(
+    navigationManager: NavigationManager,
+) {
+    Column {
+        NordicAppBar(
+            text = stringResource(id = R.string.client),
+            onNavigationButtonClick = {
+                navigationManager.navigateUp()
+            }
+        )
+
+        RequireBluetooth {
+            val clientViewModel: ClientViewModel = hiltViewModel()
+            val state by clientViewModel.state.collectAsState()
+
+            when (state) {
+                ConnectionState.Initializing -> {
+                    OutlinedCard(
```

Comment:
For readability purpose I think it would be better to extract this cases to separate functions and do
when (state) {
ConnectionState.Initializing -> InitializingView()
ConnectionState.Connecting -> ConnectingView()